build.sbt

```diff
@@ -248,6 +248,8 @@ lazy val metals = project
       // for BSP
       "org.scala-sbt.ipcsocket" % "ipcsocket" % "1.0.0",
       "ch.epfl.scala" % "bsp4j" % V.bsp,
+      "ch.epfl.scala" % "bloop-launcher_2.12" % V.bloop,
+      "ch.epfl.scala" % "bloop-launcher-core_2.12" % V.bloop,
```

Comment:
Do we need both the launcher and launcher-core?

Reply:
I think so, I had `ClassMissingException` when I didn't include core.

Reply:
You should only use `bloop-launcher`, bloop-launcher-core contains the unshaded classes. What `ClassNotFoundException` did you get?

Reply:
I removed it and currently it just freezes:
```
INFO  time: initialize in 0.2s
Starting the bsp launcher for bloop...
```

but previously it was I think `UnixDomainSocket` - for sure a class from `sockets`

Reply:
What happens if you only use `bloop-launcher-core`? I don't think we need the shaded classes, it would be good to share the coursier dependency for example

Reply:
It works with just `bloop-launcher-core`, so leave it as the single dependency?

Reply:
My vote goes to use `bloop-launcher-core` and deal with binary conflicts like we do with any other 3rd party dep

Reply:
I agree, I think it's fine from Metals perspective.

Reply:
Sounds good but keep in mind that you will then need to manage any dependency breakage that happens upstream. Not a big deal but worth thinking about

Reply:
Maybe we should start using https://github.com/scalacenter/sbt-missinglink 🤔

Reply:
https://github.com/scalameta/metals/issues/1007